### PR TITLE
chore(deps): refresh dockerode@3.3.1

### LINKED ISSUE #28725: node-gyp build error (cmake missing) for cpu-features
### Current behavior

Executing `yarn` in a local copy of the [cypress repo](https://github.com/cypress-io/cypress) on Windows 10 or 11 causes a warning on installing the optional dependency:

- [cpu-features](https://www.npmjs.com/package/cpu-features)

[node-gyp](https://www.npmjs.com/package/node-gyp) outputs the error message

> Configuring dependencies
>  'cmake' is not recognized as an internal or external command,   operable program or batch file.

and [cpu-features](https://www.npmjs.com/package/cpu-features) fails to install

### Desired behavior

[cpu-features](https://www.npmjs.com/package/cpu-features) should install without errors on Windows 10 and 11

### Test code to reproduce

On Windows 11

- Install [Python 3.11](https://devguide.python.org/versions/) from the [Microsoft Store](https://apps.microsoft.com/detail/9NRWMJP3717K?hl=en-us&gl=US).

- Install [Visual Studio Community](https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community) version 2022 (version `17.8.4`), using the workload "Desktop development with C++" with its default selections (which includes Windows 11 SDK). Select (or use Visual Studio Installer > Modify) to add Optional component Windows 10 SDK (for example `10.0.20348.0`).

```shell
git clone https://github.com/cypress-io/cypress
```

Use Node.js `18.15.0`

If testing on existing clone then

```shell
git switch develop
git clean -x -f -d
```

using git Bash shell, execute

```shell
export CI=1
yarn
```

### Cypress Version

First reported on `13.6.2` 

Reproducible on `13.6.3` commit https://github.com/cypress-io/cypress/commit/55e4898a143c96acb71f9585a410593e04f6dbad

### Node version

`v18.15.0`

### Operating System

Microsoft Windows 11 (and Windows 10)

### Debug Logs

```shell
[5/6] Building fresh packages...
warning Error running install script for optional dependency: "C:\\Users\\mikem\\github\\cypress-io\\cypress\\node_modules\\cpu-features: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments:
Directory: C:\\Users\\mikem\\github\\cypress-io\\cypress\\node_modules\\cpu-features
Output:
Configuring dependencies
  'cmake' is not recognized as an internal or external command,
  operable program or batch file.
C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\MSBuild\\Microsoft\\VC\\v170\\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'C:\\Users\\mikem\\github\\cypress-io\\cypress\\node_modules\\cpu-features\\deps\\cpu_features\\CMakeLists.txt' exited with code 9009. [C:\\Users\\mikem\\github\\cypress-io\\cypress\\node_modules\\cpu-features\\build\\config_deps.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\MSBuild\\Current\\Bin\\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\\Users\\mikem\\github\\cypress-io\\cypress\\node_modules\\@npmcli\\run-script\\node_modules\\node-gyp\\lib\\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:519:28)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:294:12)
gyp ERR! System Windows_NT 10.0.22631
gyp ERR! command \"C:\\\\Program Files\\\\nodejs\\\\node.exe\" \"C:\\\\Users\\\\mikem\\\\github\\\\cypress-io\\\\cypress\\\\node_modules\\\\@npmcli\\\\run-script\\\\node_modules\\\\node-gyp\\\\bin\\\\node-gyp.js\" \"rebuild\"
gyp ERR! cwd C:\\Users\\mikem\\github\\cypress-io\\cypress\\node_modules\\cpu-features
gyp ERR! node -v v21.6.0
gyp ERR! node-gyp -v v8.4.1
gyp ERR! not ok"
info This module is OPTIONAL, you can safely ignore this error
```

### Other

- The issue with the failure of [cpu-features](https://www.npmjs.com/package/cpu-features) to correctly install on Windows was resolved in [cpu-features@0.0.3](https://github.com/mscdex/cpu-features/releases/tag/v0.0.3) (see commit https://github.com/mscdex/cpu-features/commit/c3704d4c087c74d19f246fffe543814ca4755fd6).
- Cypress is however locked to use [cpu-features@0.0.2](https://github.com/mscdex/cpu-features/releases/tag/v0.0.2) as a transient dependency of [dockerode@3.3.1](https://github.com/apocas/dockerode/releases/tag/v3.3.1).

```text
$ yarn why cpu-features
yarn why v1.22.21
[1/4] Why do we have the module "cpu-features"...?
[2/4] Initialising dependency graph...
warning Resolution field "pretty-format@26.4.0" is incompatible with requested version "pretty-format@29.4.3"
warning Resolution field "pretty-format@26.4.0" is incompatible with requested version "pretty-format@^27.0.2"
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "cpu-features@0.0.2"
info Reasons this module exists
   - "_project_#@tooling#system-tests#dockerode#docker-modem#ssh2" depends on it
   - Hoisted from "_project_#@tooling#system-tests#dockerode#docker-modem#ssh2#cpu-features"
Done in 1.47s.
```

https://github.com/cypress-io/cypress/blob/8a02c2de8365cf0a042ac153368985d044821c37/system-tests/package.json#L58

https://github.com/cypress-io/cypress/blob/8a02c2de8365cf0a042ac153368985d044821c37/yarn.lock#L11782-L11787

### PR DESCRIPTION
**Pull request by MikeMcC399**
- Closes #28725

### Additional details

This PR allows [cpu-features](https://www.npmjs.com/package/cpu-features) to be installed on Microsoft Windows operating systems in a Visual Basic environment without error as part of executing `yarn` to install and build Cypress from source. Previously the installation failed with an error that `cmake` could not be found.

The PR removes and re-adds [dockerode](https://www.npmjs.com/package/dockerode) `3.3.1` in [system-tests/package.json](https://github.com/cypress-io/cypress/blob/develop/system-tests/package.json) to force re-evaluation of transient dependencies.

- [cpu-features@0.0.2](https://github.com/mscdex/cpu-features/releases/tag/v0.0.2) is updated to
- [cpu-features@0.0.9](https://github.com/mscdex/cpu-features/releases/tag/v0.0.9)

The newer version of `cpu-features` contains a fix which allows the npm module  [cpu-features](https://www.npmjs.com/package/cpu-features) to rebuild correctly on Windows without generating a message that `cmake` is missing (see commit https://github.com/mscdex/cpu-features/commit/c3704d4c087c74d19f246fffe543814ca4755fd6).

## Steps to test

On Microsoft Windows 11

- Install [Python 3.11](https://devguide.python.org/versions/) from the [Microsoft Store](https://apps.microsoft.com/detail/9NRWMJP3717K?hl=en-us&gl=US). Confirm that `python3 --version` shows `3.11.x`.

- Install [Visual Studio Community](https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community) version 2022, using the workload "Desktop development with C++" with its default selections (which includes Windows 11 SDK). Select (or use Visual Studio Installer > Modify) to add Optional component Windows 10 SDK (for example `10.0.20348.0`).

```shell
git clone --branch issue-28725-cpu-features-install https://github.com/MikeMcC399/cypress
cd cypress
```

Use Node.js `18.15.0`

then execute the following

```shell
yarn config set network-timeout 600000
yarn
```

No build error should be reported from `cpu-features`.

There should be no errors between the following two log lines:

```text
[5/6] Building fresh packages...
[6/6] Cleaning modules...
```

_The Yarn network-timeout setting may be necessary as a workaround for a Yarn Classic issue on Windows if Cypress has not previously been installed and there is no Yarn cache built up. Otherwise Yarn may erroneously report_
> info There appears to be trouble with your network connection. Retrying...

followed by
> ESOCKETTIMEDOUT

### How has the user experience changed?

End users see no change. This affects developers only.

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by MikeMcC399** [comment]
I successfully tested this PR locally on Windows 10 and 11.

- ~~CircleCI is failing in the [linux-x64-contributor](https://app.circleci.com/pipelines/github/cypress-io/cypress/59306/workflows/fe2b57ac-68ab-462b-9aeb-94f3cd7880f7) workflow, however the failures do not seem to be related to changes made by this PR.~~ Edit: no longer failing after rebasing. Now it is in status "On Hold".
- Snyk checks fail. It looks like these tests are not set up to run from a contributor fork. Edit: see PR https://github.com/cypress-io/cypress/pull/28093
- I'm not seeing any `windows` workflow running in [CircleCI](https://app.circleci.com/pipelines/github/cypress-io/cypress) so the relevant changes are not being tested. (I also noticed that there are no recent successful runs of the `windows` workflow in recent days.)

**Comment by jennifer-shehane** [comment]
@MikeMcC399 The windows tests run on develop or other branches that we've set up to do all of the same tests in develop. They're just so slow that we don't run them on all PRs. You can add it to run here though within the CircleCI workflow. We'll have someone review as soon as we can - they may have to set up a windows vm which takes a little bit of time though.

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

Thank you for explaining how the tests run. I don't believe that I can easily trigger the `windows` workflow in CircleCI since this would require write privileges in CircleCI that I don't have. Let me leave it with your team to review. Please let me know if there is anything more that you would like me to do.

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ _Released 1/16/2024 (PENDING)_
 - Updated [`@cypress/unique-selector`](https://www.npmjs.com/package/@cypress/unique-selector) to include a performance optimization. It's possible this could improve performance of the selector playground. Addressed in [#28571](https://github.com/cypress-io/cypress/pull/28571).
 - Replace [`CircularJSON`](https://www.npmjs.com/package/circular-json) with its successor [`flatted`](https://www.npmjs.com/package/flatted) version `3.2.9`. This resolves decoding issues observed in complex objects sent from the browser. Addressed in [#28683](https://github.com/cypress-io/cypress/pull/28683).
 - Updated [`better-sqlite3`](https://www.npmjs.com/package/better-sqlite3) from `8.7.0` to `9.2.2` to fix macOS Catalina issues. Addresses [#28697](https://github.com/cypress-io/cypress/issues/28697).
+- Refreshed dependencies of [dockerode](https://www.npmjs.com/package/dockerode) to update [cpu-features](https://www.npmjs.com/package/cpu-features) from `0.0.2` to `0.0.9`. This resolves an issue building Cypress from source on Windows operating systems. Addresses [#28725](https://github.com/cypress-io/cypress/issues/28725).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]: Since this is a development dependency update, no changelog entry is added and the PR title can be listed as `chore`.

Comment (MikeMcC399) [reply cli/CHANGELOG.md]: Thanks for your advice! I've removed the changelog entry and updated the PR title / commit to use `chore` instead of `fix`.

**Comment by MikeMcC399** [comment]
This issue is also occurring (without the changes in this PR) in CircleCI on Ubuntu when no cache has been established. See for instance https://app.circleci.com/pipelines/github/cypress-io/cypress/59096/workflows/ccbc18b9-3450-4d62-8c35-00943b1afa32/jobs/2454085.

image


The PR also fixes the problem in Ubuntu in CircleCI.

The problem appears to be masked in [GitHub runners images](https://github.com/actions/runner-images). Ubuntu and Windows runners have `cmake` installed by default.

**Review by mschile — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Review by cacieprins — APPROVED** [review]

**Comment by MikeMcC399** [comment]
I believe that this PR is now ready to merge.

https://app.circleci.com/pipelines/github/cypress-io/cypress/59354/workflows/a9e237d5-bc0e-46f3-b6b7-c8ff60107241 showed a successful CircleCI run.

I'm unsure what you will do with the branch protection rule for the required check `Snyk_SCA_Scan (18.x) `. This is now being skipped in PRs from forks, so hopefully you have permissions to override the rule?!

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

Thanks for merging! I checked locally on Windows 11 and the fix is working. I also checked locally on Ubuntu 22.04 that there is no regression. 🙂

**Comment by MikeMcC399** [comment]
- The issue in CircleCI on Ubuntu described in https://github.com/cypress-io/cypress/pull/28726#issuecomment-1894420136 is also resolved by this PR. See for instance

https://app.circleci.com/pipelines/github/cypress-io/cypress/59422/workflows/a8f6ff96-a748-409c-8414-4d349354942a/jobs/2469890 > Install Node Modules

where no errors are reported after the step 
`[5/6] Building fresh packages...`

image

**Pull request merged**